chore: add export testing to all packages

austin94 opened the pull request:
## Description

This PR adds root export testing to all packages. This is to help reduce incidents where we forget to add a new component/utility to the root export. Similar to #211 

## Detail

These tests use a `glob` utility to help with finding the components.  Since each package is unique it was difficult to add with the standard `readdir` node methods.

A few of the packages (loaders/tooltips) needed some custom logic to match the expected exports, but overall the method is consistent.

jzempel commented on packages/autocomplete/src/index.spec.js:
```diff
@@ -0,0 +1,27 @@
+
+      expect(Object.keys(rootIndex).sort()).toEqual(mappedFiles);
+      done();
+    });
```
Can't this basic `glob` structure be abstracted out to `react-testing` or `react-utilties`?

austin94 replied:
I've abstracted this into `react-testing` and added the appropriate customizations to the few packages that needed it.

austin94 commented:
> Along with the common code abstraction, consider whether it makes sense to remove the associated PR checklist item from

I've removed this checklist item and also added a new `index.spec.js` to the `.template` component.

austin94 commented:
> Along with the common code abstraction, consider whether it makes sense to remove the associated PR checklist item from

I've removed this checklist item and also added a new `index.spec.js` to the `.template` component.

ryanseddon commented on packages/testing/src/utils/matchExports.js:
```diff
@@ -0,0 +1,49 @@
+
+      const mappedFiles = fileMapper(files);
+
+      expect(keys).toEqual(mappedFiles);
```
should we really have the `expect` hidden away in another module? Can we have this method just return mapped files and then in each test we do the `expect` there.

austin94 replied:
Addressed in #228
